soh/src/code/z_sram.c

```diff
@@ -416,6 +416,95 @@ void GiveLinkDungeonReward(GetItemID getItemId) {
     }
 }
 
+void GiveLinkDungeonItem(GetItemID getItemId) {
+    int mapIndex;
+
+    switch (getItemId) {
+        case GI_DEKU_TREE_MAP:
+        case GI_DEKU_TREE_COMPASS:
+            mapIndex = SCENE_YDAN;
+            break;
+        case GI_DODONGOS_CAVERN_MAP:
+        case GI_DODONGOS_CAVERN_COMPASS:
+            mapIndex = SCENE_DDAN;
+            break;
+        case GI_JABU_JABUS_BELLY_MAP:
+        case GI_JABU_JABUS_BELLY_COMPASS:
+            mapIndex = SCENE_BDAN;
+            break;
+        case GI_FOREST_TEMPLE_MAP:
+        case GI_FOREST_TEMPLE_COMPASS:
+        case GI_FOREST_TEMPLE_SMALL_KEY:
+        case GI_FOREST_TEMPLE_BOSS_KEY:
+            mapIndex = SCENE_BMORI1;
+            break;
+        case GI_FIRE_TEMPLE_MAP:
+        case GI_FIRE_TEMPLE_COMPASS:
+        case GI_FIRE_TEMPLE_SMALL_KEY:
+        case GI_FIRE_TEMPLE_BOSS_KEY:
+            mapIndex = SCENE_HIDAN;
+            break;
+        case GI_WATER_TEMPLE_MAP:
+        case GI_WATER_TEMPLE_COMPASS:
+        case GI_WATER_TEMPLE_SMALL_KEY:
+        case GI_WATER_TEMPLE_BOSS_KEY:
+            mapIndex = SCENE_MIZUSIN;
+            break;
+        case GI_SPIRIT_TEMPLE_MAP:
+        case GI_SPIRIT_TEMPLE_COMPASS:
+        case GI_SPIRIT_TEMPLE_SMALL_KEY:
+        case GI_SPIRIT_TEMPLE_BOSS_KEY:
+            mapIndex = SCENE_JYASINZOU;
+            break;
+        case GI_SHADOW_TEMPLE_MAP:
+        case GI_SHADOW_TEMPLE_COMPASS:
+        case GI_SHADOW_TEMPLE_SMALL_KEY:
+        case GI_SHADOW_TEMPLE_BOSS_KEY:
+            mapIndex = SCENE_HAKADAN;
+            break;
+        case GI_BOTTOM_OF_THE_WELL_MAP:
+        case GI_BOTTOM_OF_THE_WELL_COMPASS:
+        case GI_BOTTOM_OF_THE_WELL_SMALL_KEY:
+            mapIndex = SCENE_HAKADANCH;
+            break;
+        case GI_ICE_CAVERN_MAP:
+        case GI_ICE_CAVERN_COMPASS:
+            mapIndex = SCENE_ICE_DOUKUTO;
+            break;
+        case GI_GANONS_CASTLE_BOSS_KEY:
+            mapIndex = SCENE_GANON;
+            break;
+        case GI_GERUDO_TRAINING_GROUNDS_SMALL_KEY:
+            mapIndex = SCENE_MEN;
+            break;
+        case GI_GERUDO_FORTRESS_SMALL_KEY:
+            mapIndex = SCENE_GERUDOWAY;
+            break;
+        case GI_GANONS_CASTLE_SMALL_KEY:
+            mapIndex = SCENE_GANONTIKA;
+            break;
+    }
```

Comment:
this block is also in `z_parameter.c`, it'd be good to move it somewhere that can be called from both places

i didn't highlight everything that's duplicated, i just think something that takes in the `GI` and spits out the `mapIndex` would be useful

Reply:
I wanted to do this, but here it uses `GI_` and in parameter it uses `ITEM_`, is there an easy way I could handle both without listing all items twice?

Reply:
oh man i missed that, yeah, that's kind of frustrating, i know it's possible but i think it'll be better to look into ways to reduce copypasta once the item table rework stuff is merged, we can leave it for now